ocean_provider/utils/test/test_currency.py

```diff
@@ -52,13 +60,37 @@ def test_to_wei():
         with localcontext(ETHEREUM_DECIMAL_CONTEXT):
             to_wei(MAX_ETHER + 1)
 
-    USDT_DECIMALS = 6
-    assert (
-        to_wei("0", USDT_DECIMALS) == 0
-    ), "Zero ether of USDT should equal zero wei of USDT"
-    assert (
-        to_wei("0.123456789123456789", USDT_DECIMALS) == 123456000_000000000
-    ), "Conversion from ether to wei using decimals failed"
-    assert (
-        to_wei("1.123456789123456789", USDT_DECIMALS) == 1_123456000_000000000
-    ), "Conversion from ether to wei using decimals failed"
+
+@pytest.mark.unit
+def test_parse_units():
+    """Test the parse_units function"""
+    assert parse_units("0", USDT_DECIMALS) == 0
+    assert parse_units("0.123456789123456789", USDT_DECIMALS) == 123456
+    assert parse_units("1.123456789123456789", USDT_DECIMALS) == 1_123456
+    assert parse_units("5278.02", USDT_DECIMALS) == 5278_020000
+    assert parse_units(MIN_USDT, USDT_DECIMALS) == MIN_WEI
+    assert parse_units(MAX_USDT, USDT_DECIMALS) == MAX_WEI
+
+    with pytest.raises(ValueError):
+        # Use ETHEREUM_DECIMAL_CONTEXT when performing arithmetic on MAX_USDT
+        with localcontext(ETHEREUM_DECIMAL_CONTEXT):
```

Comment:
I think the test is more relevant if we move the localcontext part outside (switching the `with`s), to scope out the ValueError. i.e. to outrule the possibility that localcontext itself throws the ValueError. Same goes for a few lines now, for the same usage.

Reply:
Sure this makes sense to me.  I'll make sure to include this change in the new ocean.py PR I mentioned in my comment above.